src/ngx_http_lua_directive.c

```diff
@@ -464,8 +466,27 @@ ngx_http_lua_rewrite_by_lua(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
-    if (llcf->rewrite_handler) {
-        return "is duplicate";
+    if (handlers == NULL) {
+        handlers = ngx_array_create(cf->pool, 1,
+                                    sizeof(ngx_http_lua_phase_handler_t));
+
+        if (handlers == NULL) {
+            return NGX_CONF_ERROR;
+        }
+
+        llcf->rewrite_handlers = handlers;
+    }
+
+    if (handlers->nelts >= NGX_HTTP_LUA_MAX_HANDLER_EACH_PHASE) {
+        ngx_conf_log_error(NGX_LOG_ERR, cf, 0,
+                           "the number of rewrite_by_lua* directives exceeds %d",
```

Comment:
This line exceeds 80 columns. Please use the `ngx-releng` tool to check such things. Thank you :)